tests/accelerators/test_accelerator_registry.py

```diff
@@ -0,0 +1,57 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from pytorch_lightning import Trainer
+from pytorch_lightning.accelerators import Accelerator, AcceleratorRegistry
+
+
+def test_accelerator_registry_with_new_accelerator():
+
+    accelerator_name = "custom_accelerator"
+
+    class TestAccelerator(Accelerator):
+        @staticmethod
+        def parse_devices(devices):
+            return devices
+
+        @staticmethod
+        def get_parallel_devices(devices):
+            return ["foo"] * devices
+
+        @staticmethod
+        def auto_device_count():
+            return 3
+
+        @staticmethod
+        def is_available():
+            return True
+
+        @staticmethod
+        def name():
+            return accelerator_name
```

Comment:
Now the name isn't used anywhere outside of the accelerator registry. We don't need to require it from the API/interface, which would reduce the overhead. Instead, I think we shouldoffer the register_accelerator classmethod  on the accelerator to allow for customization. In the future, we could have multiple configurations per accelerator. One example is for GPUs, and enabling inter-batch parallelism 

https://github.com/PyTorchLightning/pytorch-lightning/blob/d923dff62763b2096bb2df1b1bbfc7f8540b4bbc/pytorch_lightning/strategies/strategy.py#L445-L447

Reply:
@ananthsub Couple of questions.

1. How do you think we could have multiple of configurations? In Strategies/`register_strategies`, we have different configurations with different values being passed to class's constructor.

2. Do you think we should go ahead with having `name` as a class attribute or have it being an attribute?

Reply:
> How do you think we could have multiple of configurations? In Strategies/register_strategies, we have different configurations with different values being passed to class's constructor.

It'd be the same here. For instance, the GPU accelerator could accept a flag like `enable_inter_batch_parallelism: bool = False` to trigger this feature: https://github.com/PyTorchLightning/pytorch-lightning/blob/bc1c8b926c5072f58f42ad4b7413a8ef5c904c85/pytorch_lightning/loops/fit_loop.py#L350-L353

to me, this is no different than how someone could specify DDP find unused parameters on the DDP strategy.

> Do you think we should go ahead with having name as a class attribute or have it being an attribute?

Is name referenced anywhere outside of the registry? If it isn't, then we don't need to require this attribute at all.

Reply:
It wouldn't be a difference whether you require the name or the class-method to register it, except for the class-method to be more flexible since it allows multiple configurations :)

Reply:
After discussing offline with @kaushikb11 , we both agreed that a `register_accelerators` approach would support everything we need and be more flexible compared to the name attribute or function :)

Reply:
@justusschock @ananthsub Updated!